agimus_controller/agimus_controller/robot_model_factory/obstacle_params_parser.py

```diff
@@ -0,0 +1,201 @@
+                        )
+                else:
+                    print(f"Invalid collision pair: {pair}.")
+        else:
+            print("No collision pairs defined.")
+
+        return new_cmodel
+
+    def add_collision_pair(
+        self, cmodel: pin.GeometryModel, name_object1: str, name_object2: str
+    ) -> pin.GeometryModel:
+        if cmodel.existGeometryName(name_object1) and cmodel.existGeometryName(
+            name_object2
+        ):
+            object1_id = cmodel.getGeometryId(name_object1)
+            object2_id = cmodel.getGeometryId(name_object2)
+            cmodel.addCollisionPair(pin.CollisionPair(object1_id, object2_id))
+        else:
+            print(
+                f"Object ID not found for collision pair: {object1_id} and {object2_id}."
+            )
+        return cmodel
+
+    def transform_model_into_capsules(
+        self, model: pin.GeometryModel
+    ) -> pin.GeometryModel:
+        """Modifying the collision model to transform the spheres/cylinders into capsules which makes it easier to have a fully constrained robot."""
+        model_copy = model.copy()
+
+        # Going through all the geometry objects in the collision model
+        cylinders_name = [
+            obj.name
+            for obj in model_copy.geometryObjects
+            if isinstance(obj.geometry, Cylinder)
+        ]
+        for cylinder_name in cylinders_name:
+            basename = cylinder_name.rsplit("_", 1)[0]
+            col_index = int(cylinder_name.rsplit("_", 1)[1])
+            sphere1_name = basename + "_" + str(col_index + 1)
+            sphere2_name = basename + "_" + str(col_index + 2)
+            if not model_copy.existGeometryName(
+                sphere1_name
+            ) or not model_copy.existGeometryName(sphere2_name):
+                continue
+
+            # Sometimes for one joint there are two cylinders, which need to be defined by two capsules for the same link.
+            # Hence the name convention here.
+            capsules_already_existing = [
+                obj.name
+                for obj in model_copy.geometryObjects
+                if (basename in obj.name and "capsule" in obj.name)
+            ]
+            capsule_name = basename + "_capsule_" + str(len(capsules_already_existing))
+            geom_object = model_copy.geometryObjects[
+                model_copy.getGeometryId(cylinder_name)
+            ]
+            placement = geom_object.placement
+            parentJoint = geom_object.parentJoint
+            parentFrame = geom_object.parentFrame
+            geometry = geom_object.geometry
+            geom = pin.GeometryObject(
+                capsule_name,
+                parentFrame,
+                parentJoint,
+                Capsule(geometry.radius, geometry.halfLength),
+                placement,
+            )
+            RED = np.array([249, 136, 126, 125]) / 255
```

Comment:
It's sad to hardcode in red the collision model, we can launch a poll to decide the new color of the collision model convexified?